### ISSUE TITLE
Failed to output path.root in "terraform console"

### ISSUE THREAD

**Issue by eilgin** [hashicorp/terraform#16449]
### Terraform Version
_Using MacOS sierra 10.12.6._

Used version:
```
$ terraform --version
Terraform v0.10.7
```

### Crash Output
https://gist.github.com/eilgin/9110fc9755e0f2719d99c1c40ad76409

### Expected Behavior
I expect the console to return the value of `path.{cwd,root}`.

### Actual Behavior
It crashes for `path.root` but **not** for `path.cwd`:
```
> path.cwd
/Users/foo/bar
> path.root
panic: runtime error: invalid memory address or nil pointer dereference
...
```

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform console`
2. `path.root`

### Important Factoids
ø

### References
I didn't find anything relevant with the following filters: `is:issue is:open console path root`

**Comment by apparentlymart** [comment]
Thanks for reporting this @eilgin, and sorry for the crash.

The evaluation environment for `terraform console` is a bit different than a normal config, but indeed this is _intended_ to work. There's probably a bit of missing context in how the interpolator is set up in `console` where the configuration isn't fully loaded -- since most things `terraform console` deals with are from state rather than config.

`terraform console` will be getting a bit of a revamp soon as part of our configuration language improvements (since `console` expressions are configuration language expressions) and so while we're looking at that we can take a look at what's missing here and get it fixed.

**Comment by apparentlymart** [comment]
Hi again @eilgin!

I'm pleased to report that this does indeed seem to be fixed in v0.12.0-alpha1, and so will be fixed in the forthcoming v0.12.0 final release:

```
> path.root
/home/apparentlymart/test
> path.cwd
/home/apparentlymart/test
> path.module
/home/apparentlymart/test
```

`terraform console` now begins by performing the same preparation steps that are done during other commands like `terraform plan`, which means that all of the usual named objects should be available exactly as they would be in configuration files in the root module.

Since this is now fixed in the master branch, I'm going to close this out. Thanks again for reporting this, and sorry for the delay in addressing it.
